File: backend/app/models/activity_log.py
from sqlalchemy import Column, DateTime, Integer, String, Text, func
from app.core.database import Base


class ActivityLog(Base):
    __tablename__ = "activity_logs"

    id = Column(Integer, primary_key=True, index=True)
    event_type = Column(String(100), nullable=False)
    entity_type = Column(String(100), nullable=False)
    entity_id = Column(Integer, nullable=True)
    severity = Column(String(20), nullable=False, default="info")
    message = Column(Text, nullable=False)
    details_json = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)

File: frontend/src/types/activityLog.ts
export type ActivityLog = {
  id: number;
  event_type: string;
  entity_type: string | null;
  entity_id: string | number | null;
  severity: string;
  message: string | null;
  details_json: string | null;
  created_at: string;
};

export type ActivityLogFilterParams = {
  event_type?: string;
  entity_type?: string;
  severity?: string;
  limit?: number;
  offset?: number;
};
